pkg/clusters/addons/kong/kong.go

```diff
@@ -296,3 +387,74 @@ func urlForService(ctx context.Context, cluster clusters.Cluster, nsn types.Name
 
 	return nil, fmt.Errorf("service %s has not yet been provisoned", service.Name)
 }
+
+// deployKongEnterpriseLicenseSecret deploy secret using license json data
+func deployKongEnterpriseLicenseSecret(ctx context.Context, cluster clusters.Cluster, namespace, name, licenseJSON string) error {
+	newSecret := &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"license": []byte(licenseJSON),
+		},
+	}
+
+	_, err := cluster.Client().CoreV1().Secrets(namespace).Create(ctx, newSecret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating kong-enterprise-license secret, err %w", err)
+	}
+
+	return nil
+}
+
+func prepareSecrets(ctx context.Context, cluster clusters.Cluster, namespace, password string) error {
+	kubeconfig, err := utils.TempKubeconfig(cluster)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(kubeconfig.Name())
+
+	stderr := new(bytes.Buffer)
+	pwd := fmt.Sprintf("--from-literal=password=%s", password)
+	cmd := exec.CommandContext(ctx, "kubectl", "create", "secret", "generic", EnterpriseAdminPasswordSecretName, "-n", namespace, pwd, "--kubeconfig", kubeconfig.Name())
+	cmd.Stdout = io.Discard
+	cmd.Stderr = stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed creating super-admin secret %s: %w", stderr.String(), err)
+	}
+
+	pwd, err = os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed getting current dir, err %w", err)
+	}
+
+	guiF := pwd + "/secret_conf"
+
+	fi, err := os.Create(guiF)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := fi.Close(); err != nil {
+			fmt.Println("failed closing file %w", err)
+			return
+		}
+	}()
+
+	err = ioutil.WriteFile(guiF, []byte(`{"cookie_name":"04tm34l","secret":"change-this-secret","cookie_secure":false,"storage":"kong"}`), 0600)
+	if err != nil {
+		return fmt.Errorf("failed writing file admin_gui_session_conf, err %v", err)
+	}
+
+	guiFile := fmt.Sprintf("--from-file=admin_gui_session_conf=%s", guiF)
+	portFile := fmt.Sprintf("--from-file=portal_session_conf=%s", guiF)
```

Comment:
Any reason this can't use a Go client (something like build a `myAdminSecret` struct with appropriate data and then `cluster.Client().CoreV1().Secrets(namespace).Create(ctx, myAdminSecret, metav1.CreateOptions{})`) or, failing that `--from-literal` instead?

Creating and then storing the values in files seems a bit odd and imposes some extra constraints on the tool (it needs to have permission to write in its working directory). If we absolutely do need the files for some reason, this should probably create a temp file instead (which shouldn't care about permissions). However, I think you should be able to create these resources without FS interaction and/or a `kubectl` exec at all.

Reply:
Fixed and clean more duplicates https://github.com/Kong/kubernetes-testing-framework/pull/104/commits/af86fca8697a3a65c83780c926f24a709cd2864c